dept44-starter/src/main/java/se/sundsvall/dept44/configuration/WebConfiguration.java

```diff
@@ -113,6 +121,47 @@ public void extendMessageConverters(final List<HttpMessageConverter<?>> converte
 			.forEach(c -> c.getObjectMapper().setSerializationInclusion(JsonInclude.Include.NON_NULL));
 	}
 
+	@Override
+	public void addInterceptors(final InterceptorRegistry registry) {
+		var municipalityIdInterceptor = new MunicipalityIdInterceptor(allowedMunicipalityIds, municipalityIdUriIndex);
+
+		//  Add interceptor to check if the municipality ID is allowed.
+		// "/{municipalityId}" - Matches all paths where the municipality placeholder is the whole path.
+		// "/**/{municipalityId}" - Matches all paths where the municipality placeholder is the last part oft the path.
+		// "/{municipalityId}/**" - Matches all paths where the municipality placeholder is the beginning of the path.
+		// "/**/{municipalityId}/**" - Matches all paths where the municipality placeholder is in the middle of the path.
+		registry.addInterceptor(municipalityIdInterceptor)
+			.addPathPatterns("/{municipalityId}", "/{municipalityId}/**", "/**/{municipalityId}", "/**/{municipalityId}/**");
+	}
+
+	static class MunicipalityIdInterceptor implements HandlerInterceptor {
+
+		private final List<String> allowedMunicipalityIds;
+		private final Integer municipalityIdUriIndex;
+
+		public MunicipalityIdInterceptor(final List<String> allowedMunicipalityIds, final Integer municipalityIdUriIndex) {
```

Comment:
Could municipalityIdUriIndex be an int instead of an Integer?

No big issue perhaps, but this will cause unnecessary autoboxing. (Unnecessary since the variable never will/should be null here?)

Reply:
You are right, no need for Integer.